cvat-canvas3d/src/typescript/canvas3dView.ts

```diff
@@ -789,9 +816,21 @@ export class Canvas3dViewImpl implements Canvas3dView, Listener {
             loader.load(objectURL, this.addScene.bind(this));
             URL.revokeObjectURL(objectURL);
             this.dispatchEvent(new CustomEvent('canvas.setup'));
+        } else if (reason === UpdateReasons.UPDATE_CANVAS_BACKGROUND) {
+            const { canvasBackgroundColor } = this.model.data.shapeProperties;
+            this.views.top.scene.background = new THREE.Color(canvasBackgroundColor);
+            this.views.perspective.scene.background = new THREE.Color(canvasBackgroundColor);
+            this.views.side.scene.background = new THREE.Color(canvasBackgroundColor);
+            this.views.front.scene.background = new THREE.Color(canvasBackgroundColor);
+        } else if (reason === UpdateReasons.UPDATE_OPACITY) {
+            this.resetColor();
         } else if (reason === UpdateReasons.SHAPE_ACTIVATED) {
             const { clientID } = this.model.data.activeElement;
-            this.setupObjects();
+            if (clientID !== 'null') {
+                this.setupObjects(true);
```

Comment:
Why do we need to pass the argument here (based on ``this.model.data.activeElement``)? In ``this.setupObjects`` we also get access to ``this.model.data.activeElement`` (right in the first line)